(Feature) Step 3 new design

gabitoesmiapodo opened the pull request:
**Description**

Closes #1087 
Closes #1153 

**About #1153**
 
It's just what the issue says: the _"ButtonPlus"_ component is now outside _"NumericInput"_

**About #1087** 

I think the only thing worth mentioning is that the fields order is no 100% like in the designs. This is because:

a) Some of the designs were slightly outdated, and some fields were misplaced.
b) We decided to change the order of some of the fields and controls to provide a better workflow and user experience.

Those are not a big changes, but they surely will be noticed when comparing the implementation with the original designs.

**Screenshots:**

![localhost_3000_ ipad 1](https://user-images.githubusercontent.com/4015436/46889641-c1d22580-ce3a-11e8-8e3a-986b8188457a.png)
![localhost_3000_ ipad](https://user-images.githubusercontent.com/4015436/46889642-c26abc00-ce3a-11e8-8634-6747fea20230.png)
![localhost_3000_ laptop with mdpi screen 4](https://user-images.githubusercontent.com/4015436/46889643-c26abc00-ce3a-11e8-93c7-df0340d4c86f.png)
![localhost_3000_ laptop with mdpi screen 3](https://user-images.githubusercontent.com/4015436/46889644-c26abc00-ce3a-11e8-8d49-5ce62d7207e7.png)
![localhost_3000_ laptop with mdpi screen 2](https://user-images.githubusercontent.com/4015436/46889645-c26abc00-ce3a-11e8-8876-adafabab2e25.png)
![localhost_3000_ laptop with mdpi screen 1](https://user-images.githubusercontent.com/4015436/46889646-c26abc00-ce3a-11e8-80a1-8f49a57953af.png)
![localhost_3000_ ipad 2](https://user-images.githubusercontent.com/4015436/46889647-c3035280-ce3a-11e8-8a02-948391dc4e26.png)
![localhost_3000_ ipad 3](https://user-images.githubusercontent.com/4015436/46889648-c3035280-ce3a-11e8-979c-baaf4ad130df.png)
![localhost_3000_ iphone x 1](https://user-images.githubusercontent.com/4015436/46889649-c3035280-ce3a-11e8-868f-42b1b7a144e4.png)
![localhost_3000_ laptop with mdpi screen 5](https://user-images.githubusercontent.com/4015436/46889650-c3035280-ce3a-11e8-9752-b5cbe6a40f4d.png)
![localhost_3000_ iphone x](https://user-images.githubusercontent.com/4015436/46889651-c39be900-ce3a-11e8-8705-c4f1d9079d60.png)
![localhost_3000_ laptop with mdpi screen](https://user-images.githubusercontent.com/4015436/46889652-c39be900-ce3a-11e8-8181-bf277cc88474.png)

dennis00010011b commented:
@gabitoesmiapodo 
Step3: old and new styles are  mixed for fields start/end time 
latest Chrome, MacOS

<img width="470" alt="screen shot 2018-10-12 at 2 07 18 pm" src="https://user-images.githubusercontent.com/14067096/46894153-554e2b00-ce28-11e8-8d41-76d7f8c4ee15.png">

dennis00010011b commented:
@gabitoesmiapodo 
Input fields with arrows buttons: should be disabled to set invalid values  using arrows
<img width="421" alt="screen shot 2018-10-12 at 4 37 25 pm" src="https://user-images.githubusercontent.com/14067096/46898417-643fd800-ce3e-11e8-8490-e38d74bac2d9.png">

<img width="443" alt="screen shot 2018-10-12 at 4 38 10 pm" src="https://user-images.githubusercontent.com/14067096/46898421-76217b00-ce3e-11e8-8327-53892b4fb8ba.png">

dennis00010011b commented:
@gabitoesmiapodo 
Import CSV file : warning doesn't contain amount of added addresses if file contains more than 50 addresses. Warning's text should include phrase '50 addresses were added'

<img width="1440" alt="screen shot 2018-10-12 at 5 02 44 pm" src="https://user-images.githubusercontent.com/14067096/46898855-cea64780-ce41-11e8-97d2-59caf749bf43.png">

dennis00010011b commented:
@gabitoesmiapodo 
There is no placeholder for field Supply, should be 'Enter here'
<img width="1098" alt="screen shot 2018-10-12 at 5 23 32 pm" src="https://user-images.githubusercontent.com/14067096/46899062-ccdd8380-ce43-11e8-91ff-421cf1e1586c.png">

dennis00010011b commented:
@gabitoesmiapodo 
Typo. Should be 'process'

<img width="1152" alt="screen shot 2018-10-12 at 5 42 13 pm" src="https://user-images.githubusercontent.com/14067096/46899296-702f9800-ce46-11e8-9cf2-73a2dd80ec71.png">

dennis00010011b commented:
@gabitoesmiapodo 
Warning in console after each tier's addition

<img width="1440" alt="screen shot 2018-10-12 at 5 33 57 pm" src="https://user-images.githubusercontent.com/14067096/46899623-16c96800-ce4a-11e8-9fe1-d3a2ae89d724.png">

dennis00010011b commented:
@gabitoesmiapodo 
DutchL: there are no placeholders for input fields
<img width="1440" alt="screen shot 2018-10-12 at 6 13 57 pm" src="https://user-images.githubusercontent.com/14067096/46899673-bf77c780-ce4a-11e8-8901-74722abf9cb3.png">

gabitoesmiapodo commented:
@dennis00010011b 

- Step3: old and new styles are mixed for fields start/end time -> There's not really much I can do here. Native controls (arrows and such) can be disabled or hidden for Webkit browsers only, but I wouldn't recommend that. It cripples functionality.

We could use regular text inputs, but I wouldn't recommend that either: We should use date inputs for dates.

As a "fix", in addition to removing the background calendar icon when the input has focus I also removed it when the cursor hovers it. This way native controls will not overlap with the background image.

- Input fields with arrows buttons: should be disabled to set invalid values using arrows -> This should work better now.

- Import CSV file : warning doesn't contain amount of added addresses if file contains more than 50 addresses. Warning's text should include phrase '50 addresses were added' -> This is more of a programming logic issue than a styling one (which is the purpose of this issue). I created a new issue for this: https://github.com/poanetwork/token-wizard/issues/1179

- There is no placeholder for field Supply, should be 'Enter here' -> Fixed.

- Typo. Should be 'process' -> Fixed.

- Warning in console after each tier's addition -> I think this should be checked and fixed in a new issue, I created a new one here: https://github.com/poanetwork/token-wizard/issues/1180

- DutchL: there are no placeholders for input fields -> Fixed.

- Dutch - Gasprice : can't select option 'Custom' -> Ugh, I thought I got it working. It should work now.

dennis00010011b commented:
@gabitoesmiapodo 
Mincap's prompt text hidden behind button
<img width="1440" alt="screen shot 2018-10-16 at 9 46 24 pm" src="https://user-images.githubusercontent.com/14067096/47062840-3aafe500-d18d-11e8-9a16-3b6d284fb2d9.png">

dennis00010011b commented:
@gabitoesmiapodo 
Minted: should be disabled to add whitelist if field supply is empty
<img width="974" alt="screen shot 2018-10-16 at 10 28 55 pm" src="https://user-images.githubusercontent.com/14067096/47064155-1525da00-d193-11e8-80d6-9944848ef6de.png">

fernandomg commented:
@gabitoesmiapodo despite it's specified in the redesign specs that way. Can we change the 'On/Off' wording by 'Yes/No' as in the _whitelist_ option buttons?

fernandomg commented:
@gabitoesmiapodo Custom Gas Price arrows always change the decimals to `.1`

![custom-gas-price_always01](https://user-images.githubusercontent.com/3315606/47096786-4e4b6200-d206-11e8-909d-296c5cca6477.gif)

fernandomg commented on src/components/Common/ButtonPlus.js:
```diff
@@ -0,0 +1,9 @@
+import React, { Component } from 'react'
```
Same as with `ButtonCSV`

```javascript
import React from 'react'

export function ButtonPlus({ onClick, extraClassName = '' }) {
  return <div onClick={onClick} className={`sw-ButtonPlus ${extraClassName}`} />
}
```

fernandomg commented on src/components/Common/ButtonDelete.js:
```diff
@@ -0,0 +1,10 @@
+import React, { Component } from 'react'
```
Same here. I think I forgot to save the message

```javascript
import React from 'react'

export function ButtonDelete({ onClick, extraClassName = '' }) {
  return <div onClick={onClick} className={`sw-ButtonDelete ${extraClassName}`}
}
```

gabitoesmiapodo replied:
> @gabitoesmiapodo Custom Gas Price arrows always change the decimals to `.1`
> 
> ![custom-gas-price_always01](https://user-images.githubusercontent.com/3315606/47096786-4e4b6200-d206-11e8-909d-296c5cca6477.gif)

@fernandomg 

Ugh, this is a sh*tty one, let me see if I can make it work.

gabitoesmiapodo commented on src/components/Common/ButtonCSV.js:
```diff
@@ -0,0 +1,13 @@
+import React, { Component } from 'react'
+
+export class ButtonCSV extends Component {
```
Shure, I will do it.

mariano-aguero commented:
@gabitoesmiapodo 

can we remove this unused vars?

![screenshot_20181019_100815](https://user-images.githubusercontent.com/1144028/47220274-ae1d4680-d387-11e8-99f0-943bfb8dd220.png)

gabitoesmiapodo commented:
@mariano-aguero 

- can we remove this unused vars? -> Yes we can.
